### PR TITLE
feat(admin): add API call to log in, UI improvements

### PR DESCRIPTION
**Pull request by aleksei-bobrov**

### CODE REVIEW

File: Avingo/Code/UI/Admin/AdminLogIn/AdminLogInViewController.swift [hunk 1]
```diff
@@ -33,7 +34,31 @@ final class AdminLogInViewController: UIViewController {
...
 		self.logInButton.reactive.backgroundColor <~ self.viewModel.loginButtonBackgroundColor
-		self.logInButton.reactive.isEnabled <~ self.viewModel.isLoginEnabled
+		self.logInButton.reactive.isEnabled <~ self.viewModel.isLoginButtonEnabled
+		self.activityIndicator.reactive.isAnimating <~ self.viewModel.isExecuting
+		self.passwordTextField.reactive.borderColor <~ self.viewModel.passwordTextFieldBorderColor
+		self.emailAddressTextField.reactive.isEnabled <~ self.viewModel.isExecuting.map(!)
+		self.passwordTextField.reactive.isEnabled <~ self.viewModel.isExecuting.map(!)
+	}
+
+	override func viewWillAppear(_ animated: Bool) {
```
Comment (tinman-fueled) [inline Avingo/Code/UI/Admin/AdminLogIn/AdminLogInViewController.swift]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Fueled/avingo-ios/blob/665aa0b23eb51433034f8ccd16a25bda80436e90/Avingo/Code/UI/Admin/AdminLogIn/AdminLogInViewController.swift#L44"></span>Overridden methods call super Violation: Method ‘viewWillAppear(_:)’ should call to super function (overridden_super_call)</td>
    </tr>
  </tbody>
</table>

File: Avingo/Code/UI/Admin/AdminLogIn/AdminLogInViewModel.swift [hunk 2]
```diff
@@ -19,13 +19,36 @@ final class AdminLogInViewModel {
...
 			.map { return $0.0 && $0.1 }
 		self.loginButtonBackgroundColor = self.isLoginEnabled.map { $0 ? UIColor.Avingo.lightPink! : UIColor.Avingo.lightGray! }
+		self.isExecuting = Property(initial: false, then: logInAction.isExecuting.producer)
+		self.isLoginButtonEnabled = self.isLoginEnabled
+			.combineLatest(with: self.isExecuting)
+			.map { return $0.0 && !$0.1 }
+		let loginActionErrors = SignalProducer(self.logInAction.errors.map { _ in true })
+		let textFieldChanges = SignalProducer.merge(self.email.producer,
+													self.password.producer).map { _ in false }
+		self.shouldShowErrorState = Property(initial: false, then: SignalProducer.merge(textFieldChanges,loginActionErrors))
```
Comment (tinman-fueled) [inline Avingo/Code/UI/Admin/AdminLogIn/AdminLogInViewModel.swift]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Fueled/avingo-ios/blob/665aa0b23eb51433034f8ccd16a25bda80436e90/Avingo/Code/UI/Admin/AdminLogIn/AdminLogInViewModel.swift#L43"></span>Comma Spacing Violation: There should be no space before and one after any comma. (comma)</td>
    </tr>
  </tbody>
</table>

**Comment by tinman-fueled** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">make directory /Users/vagrant/Library/Developer/Xcode/DerivedData/Avingo-eulzywsbizskvlaaffgenvhxxrqh/Build/Products/Release-iphoneos/Avingo.app/Settings.bundle: File exists</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Avingo/Code/Helpers/Extensions/UIView+Avingo.swift [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	func makeRound() {
+		self.roundedCorner(withRadius: floor(min(self.bounds.width, self.bounds.height)/2))
+	}
+
+	func roundedCorner(withRadius radius: CGFloat) {
+		self.layer.cornerRadius = radius
+		self.clipsToBounds = true
+	}
+
+	func addBorder(color: UIColor, borderWidth: CGFloat = 1.0) {
```
Comment (ahujamanish) [inline Avingo/Code/Helpers/Extensions/UIView+Avingo.swift]: :+1:

**Pull request merged**